### PR TITLE
[FLINK-30730] Fix null values handling in StringIndexer and StringIndexerModel

### PR DESCRIPTION
**Pull request by Fanoid**
## What is the purpose of the change

This PR fixes null values handling in StringIndexer and StringIndexerModel.

## Brief change log

 - Skip null values in training, and handle null values in transforming.
 - Add tests for null values handling in corresponding test class.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-ml-lib/src/test/java/org/apache/flink/ml/feature/stringindexer/StringIndexerTest.java [hunk 1]
```diff
@@ -53,6 +55,9 @@ public class StringIndexerTest extends AbstractTestBase {
     private Table trainTable;
     private Table predictTable;
 
+    private Table trainTableWithNull;
```
Comment (zhipeng93) [inline flink-ml-lib/src/test/java/org/apache/flink/ml/feature/stringindexer/StringIndexerTest.java:58]: Can you simplify the test by adding null values in `trainTable` and `predictTable`? In this case, the test case would be simpler.

Moreover, it would be great if we also update the python test.

Comment (Fanoid) [reply flink-ml-lib/src/test/java/org/apache/flink/ml/feature/stringindexer/StringIndexerTest.java]: Thanks for your suggestion. I've updated the PR.  Please check the changes.

**Comment by zhipeng93** [comment]
Thanks for the update. LGTM.
